### PR TITLE
[dcmtk] Document dcmtk port usage

### LINKED ISSUE #31395: [dcmtk] unusable on windows / VS2019|VS2022
### Is your feature request related to a problem? Please describe.

vcpkg/dcmtk cannot currently be used on windows/VS2019. It currently fails with a cryptic error:

```
  foo.cxx
  D:\a\test-vcpkg-dcmtk\test-vcpkg-dcmtk\build\windows-release\vcpkg_installed\x64-windows\include\dcmtk/config/osconfig.h(1138,1): fatal error C1021: invalid preprocessor command 'errorDCMTK' [D:\a\test-vcpkg-dcmtk\test-vcpkg-dcmtk\build\windows-release\foo.vcxproj]
```

ref:
* https://github.com/malaterre/test-vcpkg-dcmtk/actions/runs/4946191041/jobs/8843930225

### Proposed solution

If I understand correctly upstream documentation:

> To reduce DCMTK's implementation complexity, C++11 shall only be used if your compiler conforms closely to the C++11 Standard. As far as we know, this is currently true for the following compilers:
> 
> GNU C++ Compiler (g++) Version 4.8.1 or later.
> Clang (clang++) Version 3.3 or later.

* https://support.dcmtk.org/docs/cxx11_support.html

the port should not add:

` -DCMAKE_CXX_STANDARD=17`

on MSVC compiler

ref:
* https://github.com/microsoft/vcpkg/blob/master/ports/dcmtk/portfile.cmake#LL32C8-L32C32

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by malaterre**
Fixes #31395.

### CODE REVIEW

File: ports/dcmtk/usage [hunk 1]
```diff
@@ -23,3 +23,8 @@ Add the following cmake code to your CMakeLists.txt to create definitions for DC
                 "DEFAULT_SUPPORT_DATA_DIR=${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/share/dcmtk/"
         )
     endif()
+
+On windows you must use specific MSVC flags:
+
+    target_compile_options(main PUBLIC "/Zc:__cplusplus")
+    set_property(TARGET main PROPERTY CXX_STANDARD 17)
```
Comment (dg0yt) [inline ports/dcmtk/usage:30]: IMO such usage requirements shall be carried by the CMake targets, not by manual instructions.

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by JonLiu1993** [comment]
@malaterre, Ping for response.

**Comment by JonLiu1993** [comment]
@malaterre, please also change vcpkg.json file, change `port-version` to 6, then run command `./vcpkg x-add-version dcmtk --overwrite-version` .

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**